text/0026-semantic-versioning.md

```diff
@@ -0,0 +1,189 @@
+| ADD Major | - Bump Major if set as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Patch | Bump Patch |
+| REMOVE Minor | - Bump Major if unset as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Major | Bump Major |
+<br/>
+
+
+#### Composite Buildpack
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Implementation buildpack major bumps | Implementation buildpack minor bumps | Implementation buildpack patch bumps |
+| Removal of an order group dropping<br>support of some workflow | Addition of an order group that do<br>not cause change in detection logic |  |
+| Swapping order of order groups |  |  |
+| Addition of an order group<br>causing change in detection logic |  |  |
+<br/>
+
+
+#### Builder
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Lifecycle major bumps | Minor bumps of components | Patch bumps of components |
+| Stack Image major bumps | Addition of a buildpack<br>as last in the order |  |
+| Buildpack major bumps |  |  |
+| Change in Stack ID |  |  |
+| Removal of a Buildpack |  |  |
+| Swapping order of buildpacks |  |  |
+| Addition of a buildpack except<br>as last in the order |  |  |
+<br/>
+
+#### Stack
+| `Major`                 | `Minor`                   | `Patch`                   |
+|-------------------------|---------------------------|---------------------------|
+| Removal of  a package   | Addition of a new package | Update of a package       |
+
+### Intial Development
+
+A version of `0.0.x` is considered to be in initial development and may not be
+suitable for user consumption.
+
+## User Policy
+
+* **Patch** bump: User should update their buildpack/builder without
+	hesitation.
+
+* **Minor** bump: User can update their buildpack/builder to avail new
+	features. Nothing will break if buildpack-set default dependency versions are
+	not overridden.
+
+* **Major** bump: Users must review changelog and test build system on update.
+	May contain breaking changes.
+
+
+### Critical security updates
+
+If a user is on the latest `Major` and `Minor` versions, critical security
+updates when they occur should be available for users on the latest `Patch`
+version.
+
+## Process
+
+This section proposes a process that will help maintainers easily determine
+whether a given change or a set of changes require a bump in the major, minor,
+or patch component.
+
+* Changes should be accepted to the main line of code only via Pull Requests.
+
+* All Pull Requests (including automated PRs by the
+  [paketo-bot](https://github.com/paketo-bot)) must be tagged with either of
+  the 3 labels:
+	* `semver/major` - denoting accepting the change will require a major
+	  version bump in the next release.
+	* `semver/minor` - denoting accepting the change will require a minor
+	  version bump in the next release.
+	* `semver/patch` - denoting accepting the change will require a patch
+	  version bump in the next release.
+
+* When releasing the next version, the maintainer must pick the most
```

Comment:
The way this is currently worded, it sounds like the onus is on maintainers to manually figure out which PRs will go into the release and manually set the correct version. It'd be great if the draft release automation did this for us. Not sure if it's overly prescriptive to lay that out here, but perhaps the project would benefit from having some codified process around this.

Seems like the [Java buildpacks](https://github.com/paketo-buildpacks/libjvm/blob/main/.github/workflows/update-draft-release.yml) are currently using [this release-drafter Github Action](https://github.com/release-drafter/release-drafter) which is highly configurable. Features include: automatic version resolution based on PR labels, automatic changelog generation.

Reply:
I'd say detailing the implementation of this process is out of scope for this RFC.